modules/helm-jhub/main.tf

```diff
@@ -30,6 +30,27 @@ resource "random_id" "jhub_proxy_token" {
   byte_length = 32
 }
 
+# ------------------------------------------------------------
+#  TLS (if needed)
+# ------------------------------------------------------------
+resource "kubernetes_secret" "tls_secret" {
+  count = var.create_tls_secret ? 1 : 0
```

Comment:
Is there a case where you won't want TLS? I think TLS should be mandatory

Reply:
For testing, you may skip TLS, but why this is "optional" is because TLS can be set up in different ways. For instance, if LetsEncrypt is used there is not need to have a secret. As it's handled differently in the helm chart. Since I envision this module remaining "general purpose" and not Brown-specific that's why I don't wanted to be mandatory

Reply:
Makes perfect sense!

Reply:
There is one part, I'm not sure how to handle though. The next resource depends on this resource, but I only want that to be the case if this resource exists... so if someone uses Let'sEncrypt, then the resource dependency needs to be different. Do you know and syntactic tricks to make that happen?

Reply:
Hrm... I don't off the top of my head. I think some folks use null resources for that sort of thing, but I'm not exactly sure how I'd structure that. Maybe take a peek at how some of the google modules are accomplishing similar things?

Reply:
So if I'm understanding you correctly, instead of using `depends_on` to conditionally make resources, use a `count` on the dependent resources. Meaning that you can use code like this:
```
resource "asdf_policy" "my_policy" {
count = kubernetes_secret != null ? 1 : 0
parameter1 = var.param1
parameter2 = var.param2
}
```

Reply:
@broestls I think I'm not following how to use the count.

The `helm_release` uses
`depends_on = [kubernetes_secret.tls_secret]` to wait for the secret. But if we are not using secrets, then I sound instead have `depends_on = [kubernetes_namespace.jhub]` . I'm not using the `depend_on` to conditionally make resource but rather what I want is to conditionally wait for a resource ....